plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObject.class.php

```diff
@@ -95,6 +111,59 @@ public function recursivelyAddInformationObjects($parentId, $totalRows, $options
     }
   }
 
+  private function updateAllFields($item, $options = array())
+  {
+    $node = new arElasticSearchInformationObjectPdo($item->id, $options);
+    $data = $node->serialize();
+
+    QubitSearch::getInstance()->addDocument($data, 'QubitInformationObject');
+
+    self::$counter++;
+    $this->logEntry($data['i18n'][$data['sourceCulture']]['title'], self::$counter);
+
+    return $node;
+  }
+
+  /**
+   * Update ES fields descendants inherit from their ancestors. This assumes QubitInformationObject atm.
+   */
+  private function updateInheritedFields($item, $options = array())
+  {
+    $node = new arElasticSearchInformationObjectPdo($item->id, $options);
+    $data = array();
+
+    if (isset($options['repository']) && null !== $repository = $node->getRepository())
+    {
+      $data['repository'] = arElasticSearchRepository::serialize($repository);
+    }
+
+    if (isset($options['inheritedCreators']))
+    {
+      foreach ($options['inheritedCreators'] as $creator)
```

Comment:
You're going only over the parent creators in this foreach. In the index and in the GUI we only inherit the parent creators if the IO doesn't have creators by itself. Check:

https://github.com/artefactual/atom/blob/qa/2.5.x/plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php#L80-L87
https://github.com/artefactual/atom/blob/qa/2.5.x/plugins/arElasticSearchPlugin/lib/model/arElasticSearchInformationObjectPdo.class.php#L1228-L1240

You should do the foreach over `$node->creators` and `$node->inheritedCreators`. Only one of then will have data after the constructor. Actually, to avoid having two foreach, you could do it over an array merge of both arrays (and change it in the link in top too, please).